app/models/subnet_domain.rb

```diff
@@ -2,9 +2,6 @@ class SubnetDomain < ActiveRecord::Base
   belongs_to :domain
   belongs_to :subnet
 
-  validates :subnet_id, :presence => true
-  validates :domain_id, :presence => true, :uniqueness => {:scope => :subnet_id}
```

Comment:
How would you create that duplication through subnet API? Either the subnet exist, then you only change its attribute, if you specify the same domain id, AR will handle that. Or it doesn't exists and you create new subnet (hence no duplication either). As stated in issue description, `SubnetDomain` object is never used directly and AFAIK there's no API to create these associations.